src/My.Extensions.Localization.Json/JsonStringLocalizer.cs

```diff
@@ -84,7 +84,7 @@ protected string GetStringSafely(string name)
             var culture = CultureInfo.CurrentUICulture;
             var resources = _resourcesCache.GetOrAdd(culture.Name, _ =>
             {
-                var resourceFile = $"{_resourceName}.{culture.Name}.json";
+                var resourceFile = (string.IsNullOrEmpty(_resourceName) ? $"{culture.Name}" : $"{_resourceName}.{culture.Name}") + ".json";
```

Comment:
Seems you are need to break the rules and make the resource file for all instead to be per resource.

Let me think if there's a better way to support this

Reply:
Not following you here.
Do yo mean to extract `var resourceFile` before `GetOrAdd` ?